### ISSUE TITLE
Meta: Support CPU hotplug

### ISSUE THREAD

**Issue by adriansr** [elastic/beats#6267]
[User reports](https://discuss.elastic.co/t/metric-beat-system-metrics/118177) that metricbeat needs to be restarted for the system module to pick up new CPUs hotplugged into the system. This is confirmed, `system.cpu.cores` does not update.

After reviewing the usage of `runtime.NumCPU` across beats, found out `libbeat/metric/system/process/process.go` also might calculate wrong CPU% if NumCPU changes. This should be reviewed.
 
To sum up:
- [ ] Fix metricbeat cores metric to detect number of cores at runtime.
- [ ] Review process CPU% calculation in libbeat for correctness if NumCPU changes.

FYI:
[CPU hotplug in VirtualBox](https://www.virtualbox.org/manual/ch09.html#cpuhotplug)

**Comment by tsg** [comment]
Is it a matter of reading the number of cores on every fetch? Or do we need to subscribe to something?

**Comment by adriansr** [comment]
For the metricbeat issue, yes, it as simple as that.
